lib/trace/log.js

```diff
@@ -0,0 +1,123 @@
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+'use strict';
+
+var _ = require('underscore');
+var util = require('util');
+
+var core = require('./core');
+
+//
+// Log forwarder
+//
+// opts = {
+//     sink: {
+//         // REQUIRED
+//         type: "log"
+//
+//         // OPTIONAL/HAS DEFAULTS
+//         level: 'info', // from [error,warn,info,debug,trace]
+//         prefix: 'trace'
+//     }
+// }
+//
+
+var logOptsDefaults = {
+    sink: {
+        type: 'log',
+        level: 'info',
+        prefix: 'trace'
+    }
+};
+
+var logLevels = {
+    error: true,
+    warn: true,
+    info: true,
+    debug: true,
+    trace: true
+}
+
+function LogTracer(ringpop, config, opts) {
+    if (!(this instanceof LogTracer)) {
+        return new LogTracer(opts);
+    }
+
+    opts.sink = _.extend(
+        {},
+        logOptsDefaults.sink,
+        config.sinkOptsDefaults.log,
+        opts.sink);
+    core.Tracer.call(this, ringpop, config, opts);
+
+    this.level = this.opts.sink.level;
+    this.prefix = opts.sink.prefix ? opts.sink.prefix + ' ' : '';
+}
+util.inherits(LogTracer, core.Tracer);
+
+
+LogTracer.validate = function validate(opts) {
+    var sinkOpts = opts.sink;
+    return (
+        core.Tracer.prototype.validate.call(this) &&
+        sinkOpts &&
+        sinkOpts.type === 'log' &&
+        logLevels[sinkOpts.level]
+    );
+};
+
+LogTracer.prototype.connectSink = function connectSink(callback) {
+    if (callback) {
+        this.once('connectSink', callback);
+    }
+    process.nextTick(this.emit.bind(this, 'connectSink'));
+};
+
+LogTracer.prototype.disconnectSink = function disconnectSink(callback) {
+    if (callback) {
+        this.once('disconnectSink', callback);
+    }
+
+    this.disconnectSource();
+    process.nextTick(this.emit.bind(self, 'disconnectSink'));
+};
+
+LogTracer.prototype.send = function send(blob, callback) {
+    if (callback) {
+        this.once('send', callback);
+    }
+
+    if (typeof blob !== 'string') { // XXX also allow Buffer?
```

Comment:
Buffer is solved by that, but what about sending plain ol' objects. I am inclined to support it directly, or to add an 'asJSON' option to the registration. Whatcha think? @jwolski